SConstruct

```diff
@@ -103,11 +103,13 @@ custom_tools = ["default"]
 
 platform_arg = ARGUMENTS.get("platform", ARGUMENTS.get("p", False))
 
-if os.name == "nt" and (platform_arg == "android" or methods.get_cmdline_bool("use_mingw", False)):
-    custom_tools = ["mingw"]
+if platform_arg == "android":
+    custom_tools = ["clang", "clang++", "as", "ar", "link"]
```

Comment:
This likely configure the OS installed clang, and might be the reason why it fails to link on OSX.
In particular, I think this configures `env["LD"]` (the linker) to the OSX/Xcode `llvm-ld`.
I would completely skip the "custom_tools" part.

Reply:
The `custom tools` is necessary for that very reason. On Windows it defaults to configuring everything for Visual Studio. However, unlike using the `mingw` tool, the `clang` tool just provides the `CC` variable, which is why all the other tools are needed too.